src/main.rs

```diff
@@ -321,6 +289,48 @@ fn main_with_result() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+/// does the actual encoding/decoding work
+fn do_work(
+    file_type: FileType,
+    verify: bool,
+    input_data: &Vec<u8>,
+    enabled_features: &EnabledFeatures,
+) -> Result<(Vec<u8>, Metrics), LeptonError> {
+    let metrics;
+    let mut output;
+
+    match file_type {
+        FileType::Jpeg => {
+            if verify {
+                (output, metrics) = encode_lepton_verify(input_data, enabled_features)?;
+            } else {
+                let mut reader = Cursor::new(input_data);
+                output = Vec::with_capacity(input_data.len());
+                let mut writer = Cursor::new(&mut output);
+
+                metrics = encode_lepton(&mut reader, &mut writer, enabled_features)?
+            }
+            let output_len = output.len();
+
+            info!(
+                "compressed input {0}, output {1} bytes (ratio = {2:.1}%)",
```

Comment:
It is not ratio per se, it is shrinkage or something like it. Maybe make it a true ratio by `(output_len as f64) / (input_data.len() as f64) * 100.0`?